### PR TITLE
[6-42] UI: Coach and Firebase notification

### PR DESCRIPTION
**Pull request by HrebenYury**
1. Added notifications in create components;
2. Added notification in the speaking component. When we have some troubles with Firebase is shows in notification;
3. Improved audio-cloud service. Added reject in Promise.

### CODE REVIEW

File: exalearn-ui/src/app/components/new-listening/new-listening.component.ts [hunk 1]
```diff
@@ -62,17 +65,23 @@ export class NewListeningComponent implements OnInit {
...
+				this.loadAudio = false;
+			});
 
 		this.audioService.getPercentage().subscribe(
 			(percentage) => {
 				this.percentage = Math.round(percentage ? percentage : 0);
 			},
 			(error) => {
-				console.log(error);
+				this.loadAudio = false;
```
Comment (vstrabko) [inline exalearn-ui/src/app/components/new-listening/new-listening.component.ts:84]: Probably worth adding user notification popup here.

Comment (vstrabko) [reply exalearn-ui/src/app/components/new-listening/new-listening.component.ts:84]: Other than that - happy to approve

File: exalearn-ui/src/app/components/new-grammar/new-grammar.component.ts [hunk 2]
```diff
@@ -38,7 +44,7 @@ export class NewGrammarComponent implements OnInit {
 	}
 
 	submit(): void {
-		this.ncService.setIsCorrectProperty(this.form);
+		/* this.ncService.setIsCorrectProperty(this.form);
```
Comment (YanaBr) [inline exalearn-ui/src/app/components/new-grammar/new-grammar.component.ts]: Do we really need to store this comment? It's better to remove all unnecessary comments

File: exalearn-ui/src/app/components/new-grammar/new-grammar.component.ts [hunk 3]
```diff
@@ -50,24 +56,27 @@ export class NewGrammarComponent implements OnInit {
...
 				.postRequest(API_PATH.NEW_GRAMMAR, this.form.value)
 				.then(() => {
+					this.notificationService.successPopUp();
 					this.grammarForm.resetForm();
-					this.load = false;
 				})
-				.catch(() => console.log('Sorry something wrong((('));
+				.catch(() => this.notificationService.errorPopUp('Sorry. Something went wrong'))
+				.finally(() => {
+					this.load = false;
```
Comment (YanaBr) [inline exalearn-ui/src/app/components/new-grammar/new-grammar.component.ts:59]: So, do we want to make `this.load = false` in every case? Or do we want to do this only on error falls?

File: exalearn-ui/src/app/components/new-listening/new-listening.component.ts [hunk 4]
```diff
@@ -134,16 +143,17 @@ export class NewListeningComponent implements OnInit {
 			//TODO
 			//Delet setTimeout above and uncomment code below. When server is ready
 
-			/* 
-			void this.apiServise
+			/* void this.apiServise
```
Comment (YanaBr) [inline exalearn-ui/src/app/components/new-listening/new-listening.component.ts]: Again long strange comments

File: exalearn-ui/src/app/components/new-topic/new-topic.component.ts [hunk 5]
```diff
@@ -58,11 +64,12 @@ export class NewTopicComponent implements OnInit {
...
+				this.notificationService.successPopUp();
 				this.form.resetForm();
-				this.load = false;
 			})
 			.catch(() => {
-				this.load = false;
-			});
+				this.notificationService.errorPopUp('Sorry. Something went wrong');
+			})
+			.finally(() => (this.load = false));
```
Comment (YanaBr) [inline exalearn-ui/src/app/components/new-topic/new-topic.component.ts:73]: Same questions as above

File: exalearn-ui/src/app/components/speaking/speaking.component.ts [hunk 6]
```diff
@@ -104,21 +111,29 @@ export class SpeakingComponent implements OnInit {
...
-			};
-			this.submit.addData('speaking', speakingAnswer);
-		});
+		await this.audioStorage
+			.uploadAudio(file, environment.cloudSpeaking)
+			.then((url) => {
+				this.audioUrlCloud = url;
+				this.isRecordReadySpinner = false;
+				this.recording = false;
+				const speakingAnswer: testAnswer = {
```
Comment (YanaBr) [inline exalearn-ui/src/app/components/speaking/speaking.component.ts]: I think we can split this part to a separate method, which create necessary data object

File: exalearn-ui/src/app/services/audio-cloud.service.ts [hunk 7]
```diff
@@ -21,23 +21,33 @@ export class AudioCloudService {
...
-					finalize(() => {
-						this.storageRef.getDownloadURL().subscribe((downloadURL: string) => {
-							resolve(downloadURL);
-						});
-					})
-				)
-				.subscribe();
+
+		return new Promise((resolve, reject) => {
+			this.storageRef = this.storage.ref(filePath);
```
Comment (YanaBr) [inline exalearn-ui/src/app/services/audio-cloud.service.ts:29]: Again, I think we can refactor it a bit, looks bulky

File: exalearn-ui/src/app/components/new-grammar/new-grammar.component.ts (hunk 3 above)
Comment (HrebenYury) [reply exalearn-ui/src/app/components/new-grammar/new-grammar.component.ts:59]: We use it while data sending to the server. This property uses for the progress bar. If data is sent ok. We stop the progress bar. If the data sent is not ok, we also stop the progress bar.

File: exalearn-ui/src/app/components/new-topic/new-topic.component.ts (hunk 5 above)
Comment (HrebenYury) [reply exalearn-ui/src/app/components/new-topic/new-topic.component.ts:73]: We use it while data sending to the server. This property uses for the progress bar. If data is sent ok. We stop the progress bar. If the data sent is not ok, we also stop the progress bar.

File: exalearn-ui/src/app/components/new-listening/new-listening.component.ts (hunk 1 above)
Comment (HrebenYury) [reply exalearn-ui/src/app/components/new-listening/new-listening.component.ts:84]: ok

File: exalearn-ui/src/app/components/speaking/speaking.component.ts (hunk 6 above)
Comment (HrebenYury) [reply exalearn-ui/src/app/components/speaking/speaking.component.ts]: ok

**Review by YanaBr — APPROVED** [review]

**Review by GeranSkot — APPROVED** [review]

**Review by HenadzV — APPROVED** [review]

**Pull request merged**